src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -267,6 +268,21 @@
             TokenTypes.BXOR_ASSIGN,
         }).collect(Collectors.toSet()));
 
+    /** List of tokens from {@link #getClassFrameWhereViolationIsFound(DetailAST)} method.*/
+    private static final List<Integer> EXPR_TOKENS = Arrays.asList(
+            TokenTypes.PLUS,
+            TokenTypes.MINUS,
+            TokenTypes.EQUAL,
+            TokenTypes.NOT_EQUAL,
+            TokenTypes.STAR,
+            TokenTypes.DIV,
+            TokenTypes.MOD,
+            TokenTypes.LT,
+            TokenTypes.GT,
+            TokenTypes.LE,
+            TokenTypes.GE
```

Comment:
What about left / right shift?

Reply:
Yeap, I forgot some of the operations (not only these)
I added missing ones and tests

Reply:
This is why I didn't like a hardcoded list. It will be easy to miss some operators.
Why can't we check all expressions?

Reply:
How, for example?